sbn/codingconv/CodingConventions.md

```diff
@@ -133,10 +133,10 @@ even at the cost of some additional key strokes, and then to minimize
 possible name collisions or ambiguities.
 
 **[R]** Use of a descriptive control variable is **required**
-  in any loop longer than five lines.
+  in any loop longer than five lines, unless implementing a mathematical formula (matrix operations?) where i,j,k indices are traditional
   
 **[D]** Declaration of identifiers starting with an underscore is **discouraged**
-  (even to denote private class members); 
+  (even to denote private class members); TODO: how about in to name constructor arguments that alias class members? I guess Foo::Foo(int _x) : x(_x) {} in this case
```

Comment:
I use the same nam**WHHAAAATT??!?** err... yes.
```cpp
Foo::Foo(int x): x{ x } {}
```
I am not going to write _that_ in a recommendation. Anyway, underscore names have earned the fame of being provate class members, so even `_x` is somehow deceiving. Sometimes I use a Python solution (`x_`), but I am not particularly proud of that neither. I would be ok to grant an exception for a short-scoped variable like in the example.
What do you think?

Reply:
haha. I'm surprised that works. I know from sad experience that `int x = x;` is valid syntax and initializes `x` with garbage.

Reply:
What about this other answer to your objection: only private data members should need initialisation in constructors.
Heh?
I am actually not confident of enunciating this as a universal law, but I believe it's reliable in most cases.

Reply:
I normally write this pattern for
`struct Pt{float x, y; Pt(float _x, float _y) : x(_x), y(_y) {}};`

Reply:
Are you trying to force the users to always fully initialise the point? and is it your explicit intention to disable a default constructor?
Otherwise, `Pt p { 8.0f, 9.0f };` should work without the constructor you described:
```cpp
struct Pt { float x, y; };
```

I don't want to focus too much on the specific example, but the questions are general.

Reply:
Partly it's because I'm an old fuddy-duddy who prefers parenthesis initialization to brace initialization. But also, you don't have to get much more complicated before the ability to have a default parameter, or two constructors taking different arguments, becomes useful.